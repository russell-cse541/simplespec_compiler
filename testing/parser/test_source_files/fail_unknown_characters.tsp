// There is a stray character on line 22, line 13, and line 31 

namespace  Main;

using Locale;
using Big;

scalar ID = ASCIIString {
    requires: value.length();
}

scalar Apple = ASCIIString {
    requires: component.juice(*);
    requires: component.seed();
}


model User {
    id: ID;
    name: String;
    region: Locale::Country;
    team: #List<User,Hey>;
    fan: Bad<Yes>;
}

const a:Bool = true;
const sillString:String = 'Yes This is a very Simple String!';

function checkit(): Bool {
    let b:Bool = true;
    let a:String = a.$length();
    return u.a.length(pred(a:String): Bool => true);
}

function checkit(a:String, b:Bool, c:SomethingFake): Bool {
    return !(5>8) || (!(a || b && c));
}