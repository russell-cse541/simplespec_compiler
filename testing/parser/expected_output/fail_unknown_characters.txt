
Unrecognized character: *  => Line 13 Col 31
Error: syntax error line: 13
Error: [1;31m Improper Expression [0m line: 13
Error: syntax error line: 14
Error: [1;31m Improper Expression [0m line: 14
Error: syntax error line: 15
Error: [1;31m Improper Expression [0m line: 19
Error: syntax error line: 20
Error: [1;31m Improper Expression [0m line: 20
Error: syntax error line: 21
Error: [1;31m Improper Expression [0m line: 21
Error: syntax error line: 22

Unrecognized character: #  => Line 22 Col 11
Error: [1;31m Improper Expression [0m line: 22
Error: syntax error line: 23
Error: [1;31m Improper Expression [0m line: 23
Error: syntax error line: 24
Error: [1;31m Improper Expression [0m line: 26
Error: syntax error line: 27
Error: [1;31m Improper Expression [0m line: 27
Error: syntax error line: 29
Error: [1;31m Improper Expression [0m line: 30
Error: syntax error line: 31

Unrecognized character: $  => Line 31 Col 22
Error: [1;31m Improper Expression [0m line: 31
Error: syntax error line: 32
Error: [1;31m Improper Expression [0m line: 32
Error: syntax error line: 33
Error: [1;31m Improper Expression [0m line: 36
Error: syntax error line: 37


[0;33m There were errors parsing the document. Start by fixing the initial error.  Other errors may be uncovered as some are corrected.[0m

